### PR TITLE
[Pluggable storage] storage factory and wrapper validation

### PR DESCRIPTION
**Pull request by EmilianoSanchez**
# Javascript commons library

## What did you accomplish?

- Added factory function that validates given wrapper and instantiates a pluggable storage.
- Additionally:
  - updated some internal event values to work with JS SDK. 
  - updated index.js to export storage factory modules and make them easier to import from an external package.

## How do we test the changes introduced in this PR?

- UTs

## Extra Notes

### CODE REVIEW

File: src/storages/pluggable/index.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  wrapper: ICustomStorageWrapper
+}
+
+/**
+ * Validate pluggable storage factory options.
+ *
+ * @param options user options
+ * @throws Will throw an error if the options are invalid. Example: wrapper is not provided or doesn't have some methods.
+ */
+function validatePluggableStorageOptions(options: any) {
```
Comment (emilianofant-split) [inline src/storages/pluggable/index.ts:24]: Missing return type declaration (`void` I guess)

File: src/storages/pluggable/index.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * Pluggable storage factory for consumer server-side & client-side SplitFactory.
+ */
+export function PluggableStorage(options: PluggableStorageOptions) {
+
+  validatePluggableStorageOptions(options);
+
+  const prefix = options.prefix ? options.prefix + '.SPLITIO' : 'SPLITIO';
+
+  return function PluggableStorageFactory({ log, metadata, readinessManager }: IStorageFactoryParams): IStorageAsync {
+
```
Comment (emilianofant-split) [inline src/storages/pluggable/index.ts:42]: Remove this empty line

File: src/storages/pluggable/index.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  const wrapper = options.wrapper;
+  const missingMethods = METHODS_TO_PROMISE_WRAP.filter(method => typeof wrapper[method] !== 'function');
+  if (missingMethods.length) throw new Error(`Wrapper instance must implement the following methods: ${missingMethods}`);
+}
+
+/**
+ * Pluggable storage factory for consumer server-side & client-side SplitFactory.
+ */
+export function PluggableStorage(options: PluggableStorageOptions) {
```
Comment (emilianofant-split) [inline src/storages/pluggable/index.ts:35]: Missing return type definition

**Review by emilianofant-split — APPROVED** [review]
Small comments, but LGTM!

File: src/storages/pluggable/index.ts (hunk 2 above)
Comment (EmilianoSanchez) [reply src/storages/pluggable/index.ts:42]: done! something else to add to the linter

**Pull request merged**